test/test-utils.js

```diff
@@ -1,23 +1,54 @@
 const fs = require('fs');
-const path = require('path');
-const {execSync} = require('child_process');
+const util = require('util');
+const glob = require('glob');
+const exec = util.promisify(require('child_process').exec);
+
+const STRIP_CHUNKHASH = /([\w\-]+\-)[a-z0-9]{8}(\.js)/g;
+const STRIP_REV = /\?rev=\w+/gm;
+const STRIP_WHITESPACE = /((\s+$)|((\\r\\n)|(\\n)))/gm;
+const UTF8_FRIENDLY_EXTS = ['css', 'html', 'js', 'jsx', 'ts', 'tsx', 'svelte', 'svg', 'vue']; // only read non-binary files (add more exts here as needed)
 
 /** setup for /tests/build/* */
-function setupBuildTest(cwd) {
-  execSync('yarn testbuild', {cwd});
+async function setupBuildTest(cwd) {
+  const results = await exec('yarn testbuild', {cwd});
+  return results;
 }
 exports.setupBuildTest = setupBuildTest;
 
-/** take an array of files, return contents of files (won’t read entire directory for performance) */
-function readFiles(files, {cwd}) {
-  if (!cwd) throw new Error(`cwd option missing, ex: readFiles(files, { cwd: __dirname })`);
+/** take a folder of files, return contents of files (won’t read entire directory for performance) */
+async function readFiles(directory, {ignore} = {}) {
+  if (!directory) throw new Error(`must specify directory`);
 
   const contents = {};
-  (Array.isArray(files) ? files : [files]).forEach((f) => {
-    const relativePath = f.replace(/^\/?/, '/');
-    const filepath = path.join(cwd, ...f.split('/'));
-    contents[relativePath] = fs.readFileSync(filepath, 'utf-8');
+  const allFiles = glob.sync(directory + `/**/*.{${UTF8_FRIENDLY_EXTS.join(',')}}`, {
+    nodir: true,
+    ignore,
   });
+
+  await Promise.all(
+    allFiles.map(async (filepath) => {
+      const relativePath = filepath.replace(directory, '').replace(/^\/?/, '/');
+      contents[relativePath] = await fs.promises.readFile(filepath, 'utf-8');
+    }),
+  );
   return contents;
 }
 exports.readFiles = readFiles;
+
+/** strip chunk hash from URLs */
+function stripChunkhash(code) {
+  return code.replace(STRIP_CHUNKHASH, '$1XXXXXXXX$2');
+}
+exports.stripChunkhash = stripChunkhash;
+
+/** strip ?rev= from URLs */
+function stripRev(code) {
+  return code.replace(STRIP_REV, '?rev=XXXXXXXXXX');
+}
+exports.stripRev = stripRev;
+
+/** strip whitespace */
+function stripWS(code) {
+  return code.replace(STRIP_WHITESPACE, '');
+}
```

Comment:
Old utils from `test/build/build.test.js` were added here (these were the only functions, but if we need any other inline logic we could always add that here too, as we need it)